feat(red): persistence MTA-3598 (#2081)
scaleway/docs-content: managed-databases/redis/reference-content/ensuring-data-persistence.mdx

Reviewer: Learn more about how to ensure data persistence in Managed Database for Redis™.

---
meta:
  title: Ensuring data persistence
  description: Learn more about how to ensuring data persistence in Managed Database for Redis™.
content:
  h1: Ensuring data persistence
  paragraph: Learn more about how to ensuring data persistence in Managed Database for Redis™.
tags: databases user redis username password
categories:
  - managed-databases
  - redis
---

> Ensuring data persistence is ensuring that the data from your applications is saved, in memory or on a disk, even when there are no ongoing sessions where the data is treated. The method used to preserve data between sessions, and avoid data loss varies depending on the compute resource you use to treat your data. It also depends on the way your data is stored.

Reviewer: Data persistence ensures that the data from your applications is saved in memory or on a disk...

> Ensuring data persistence is ensuring that the data from your applications is saved, in memory or on a disk, even when there are no ongoing sessions where the data is treated. The method used to preserve data between sessions, and avoid data loss varies depending on the compute resource you use to treat your data. It also depends on the way your data is stored.

Author: ```suggestion
Data persistence ensures that the data from your applications is saved, in memory or on a disk, even when there are no ongoing sessions where the data is treated. The method used to preserve data between sessions, and avoid data loss varies depending on the compute resource you use to treat your data. It also depends on the way your data is stored.
```

When data is stored in memory, and no data persistence strategy is put in place, the data will be gone any time the compute process is is stopped.

> To avoid this, Redis™ innately supports persistence by performing [snapshots](https://redis.io/docs/management/persistence/).

Reviewer 2: ```suggestion
To avoid this, Redis™ natively supports persistence by performing [snapshots](https://redis.io/docs/management/persistence/).
```

Reviewer 2: Preferential, but "natively" is more common in IT context

---
meta:
  title: Ensuring data persistence
  description: Learn more about how to ensuring data persistence in Managed Database for Redis™.
content:
  h1: Ensuring data persistence
  paragraph: Learn more about how to ensuring data persistence in Managed Database for Redis™.
tags: databases user redis username password
categories:
  - managed-databases
  - redis
---

Ensuring data persistence is ensuring that the data from your applications is saved, in memory or on a disk, even when there are no ongoing sessions where the data is treated. The method used to preserve data between sessions, and avoid data loss varies depending on the compute resource you use to treat your data. It also depends on the way your data is stored.

When data is stored in memory, and no data persistence strategy is put in place, the data will be gone any time the compute process is is stopped.

> To avoid this, Redis™ inately supports persistence by performing [snapshots](https://redis.io/docs/management/persistence/).

Reviewer 3: ```suggestion
To avoid this, Redis™ innately supports persistence by performing [snapshots](https://redis.io/docs/management/persistence/).
```

not sure if "innately" is easy to understand, though

This feature is supported and managed by Scaleway with Managed Databases for Redis™.

Furthermore, you can leverage the [cluster mode](faq/databases-for-redis/#what-is-the-logic-behind-the-cluster-mode) to ensure the durability of your data.

<Message type="important">
  Managed Database for Redis™ remains an in-memory database solution, and is recommended for caching use cases. The data persistence solutions presented on this page cannot guarantee data permanence.
</Message>

## Snapshots

Snapshots are available by default with a fixed configuration.

When you create a new Redis™ Database Instance, a Redis™ snapshot is also automatically created within specified intervals. These are:

- 900 seconds if one key has been updated
- 300 seconds if 10 keys have been updated
- 60 seconds if 10 000 keys have been updated

<Message type="important">

>   The snapshot configuration is managed by Scaleway and cannot be edited. If you feel this configuration does not represent your use case, please let us know via the #databases channel of the [Scaleway Slack community](https://scaleway-community.slack.com/).

Reviewer 3: ```suggestion
  The snapshot configuration is managed by Scaleway and cannot be edited. If you feel this configuration does not represent your use case, please let us know via the #databases channel of the [Scaleway Slack community](https://slack.scaleway.com/).
```

You can't signup if you use direct slack.com link

</Message>

## Cluster mode

If you want to benefit from fault-tolerant architecture in some use cases, we recommend you deploy Redis™ in [cluster mode](/faq/databases-for-redis/#what-is-the-logic-behind-the-cluster-mode), where each primary node is deployed with a replica node.

> A Redis cluster contains a minimum of 3 nodes and up to 6 nodes. Each node contains a source and a replica. The cluster nodes use hash partitioning to split the keyspace into key slots. Each replica copies the data of a specific source and can be reassigned to replicate another source or be elected as a source node if needed. This is recommended for scaling as the operation is spread across multiple nodes instead of having a single entry point.

Reviewer 3: ```suggestion
A Redis cluster contains a minimum of three nodes and up to six nodes. Each node contains a source and a replica. The cluster nodes use hash partitioning to split the keyspace into key slots. Each replica copies the data of a specific source and can be reassigned to replicate another source or be elected as a source node if needed. This is recommended for scaling as the operation is spread across multiple nodes instead of having a single entry point.
```
